packages/bot-web-ui/src/components/self-exclusion/self-exclusion.jsx

```diff
@@ -85,8 +85,8 @@ const SelfExclusionForm = props => {
         });
 
         decimal_limit.forEach(item => {
-            const amount_decimal_array = values[item].toString().split('.')[1];
-            const amount_decimal_places = amount_decimal_array ? amount_decimal_array.length || 0 : 0;
+            const amount_decimal_array = values[item]?.toString().split('.')[1];
+            const amount_decimal_places = amount_decimal_array ? amount_decimal_array.length : 0;
```

Comment:
I believe that this additional condition can be removed because:

When selecting a suitable value, when `amount_decimal_array` is array (a true value), `amount_decimal_array.length` should be false; in this case, it should be `0`.

By taking the value `'123.'` (a number with a decimal point but no digits after it), where the result of `amount_decimal_array.length` is `0`, I realized that the first operand will not be returned because, in this case, it is **not an array**.
**Code:**
`const amount_decimal_places = amount_decimal_array ? amount_decimal_array.length || 0 : 0;`
**Example:**
`const amount_decimal_array = '123.'?.toString().split('.')[1];`
// amount_decimal_array = '', typeof amount_decimal_array = string, !!amount_decimal_array = false, amount_decimal_array.length = 0
**Therefore:**
`const amount_decimal_places = '' ? 0 || 0 : 0;`
`const amount_decimal_places = false ? 0 || 0 : 0; `// here returns the latest operand
**That is, there is no case where the highlighted value in this expression is returned:**
const amount_decimal_places = amount_decimal_array ? amount_decimal_array.length || **0** : 0;
=>
The explanation suggests that the ternary expression can be simplified without losing functionality because the highlighted condition is unnecessary.